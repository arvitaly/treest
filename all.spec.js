"use strict";
var __awaiter = (this && this.__awaiter) || function (thisArg, _arguments, P, generator) {
    return new (P || (P = Promise))(function (resolve, reject) {
        function fulfilled(value) { try { step(generator.next(value)); } catch (e) { reject(e); } }
        function rejected(value) { try { step(generator["throw"](value)); } catch (e) { reject(e); } }
        function step(result) { result.done ? resolve(result.value) : new P(function (resolve) { resolve(result.value); }).then(fulfilled, rejected); }
        step((generator = generator.apply(thisArg, _arguments || [])).next());
    });
};
Object.defineProperty(exports, "__esModule", { value: true });
const fs_1 = require("fs");
const with_error_1 = require("with-error");
const Treest_1 = require("./Treest");
let treest;
beforeEach(() => {
    with_error_1.default(() => fs_1.unlinkSync(__dirname + "/__treest__/__fixtures__/module1.json"));
    with_error_1.default(() => fs_1.unlinkSync(__dirname + "/__treest__/__fixtures__/module2.json"));
    with_error_1.default(() => fs_1.unlinkSync(__dirname + "/__treest__/__fixtures__/module3.json"));
    with_error_1.default(() => fs_1.unlinkSync(__dirname + "/__treest__/__fixtures__/ClassA.json"));
    treest = new Treest_1.default({
        mode: process.argv.indexOf("-u") > -1 ? "update" : "",
        mocks: {
            fs: () => {
                return {
                    readFileSync: (name) => {
                        if (name === "./dot.txt") {
                            return ".";
                        }
                        return ",";
                    },
                };
            },
        },
    });
});
it("hello, world", () => __awaiter(this, void 0, void 0, function* () {
    const { hello } = treest.require("./__fixtures__/module1");
    expect(yield hello("Hello", "world")).toBe("Hello, world!");
}));
it("hello, John", () => __awaiter(this, void 0, void 0, function* () {
    const { hello } = treest.require("./__fixtures__/module1");
    expect(yield hello("Hello", "John")).toBe("Hello, John!");
}));
afterEach(() => {
    require.cache = {};
    expect(JSON.parse(fs_1.readFileSync(__dirname + "/__treest__/__fixtures__/module1.json").toString())).toMatchSnapshot();
    expect(JSON.parse(fs_1.readFileSync(__dirname + "/__treest__/__fixtures__/module2.json").toString())).toMatchSnapshot();
    expect(JSON.parse(fs_1.readFileSync(__dirname + "/__treest__/__fixtures__/module3.json").toString())).toMatchSnapshot();
    expect(JSON.parse(fs_1.readFileSync(__dirname + "/__treest__/__fixtures__/ClassA.json").toString())).toMatchSnapshot();
});
