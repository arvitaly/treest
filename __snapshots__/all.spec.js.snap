// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`hello, John 1`] = `
Object {
  ".hello": Array [
    Object {
      "args": Array [
        "Hello",
        "John",
      ],
      "isPromise": true,
      "result": "Hello, John!",
    },
  ],
}
`;

exports[`hello, John 2`] = `
Object {
  ".addExclamation": Array [
    Object {
      "args": Array [
        "Hello, John",
      ],
      "isPromise": false,
      "result": "Hello, John!",
    },
  ],
  ".concat": Array [
    Object {
      "args": Array [
        Array [
          "Hello",
          "John",
        ],
        Object {
          "__$__": "class",
          "exportPath": ".default",
          "moduleName": "__fixtures__/ClassA",
          "state": "{\\"value\\":\\"value1\\"}",
        },
      ],
      "isPromise": false,
      "result": "Hello, John",
    },
  ],
}
`;

exports[`hello, John 3`] = `
Object {
  ".apply": Array [
    Object {
      "args": Array [
        "Hello, John",
        Object {
          "__$__": "function",
          "exportPath": ".addExclamation",
          "moduleName": "__fixtures__/module2",
        },
      ],
      "isPromise": false,
      "result": "Hello, John!",
    },
  ],
}
`;

exports[`hello, John 4`] = `
Object {
  ".default.getSeparator": Array [
    Object {
      "args": Array [
        Object {
          "__$__": "class",
          "exportPath": ".default",
          "moduleName": "__fixtures__/ClassA",
          "state": "{\\"value\\":\\"value1\\"}",
        },
        "comma",
      ],
      "isPromise": false,
      "result": ",",
    },
  ],
}
`;

exports[`hello, world 1`] = `
Object {
  ".hello": Array [
    Object {
      "args": Array [
        "Hello",
        "world",
      ],
      "isPromise": true,
      "result": "Hello, world!",
    },
  ],
}
`;

exports[`hello, world 2`] = `
Object {
  ".addExclamation": Array [
    Object {
      "args": Array [
        "Hello, world",
      ],
      "isPromise": false,
      "result": "Hello, world!",
    },
  ],
  ".concat": Array [
    Object {
      "args": Array [
        Array [
          "Hello",
          "world",
        ],
        Object {
          "__$__": "class",
          "exportPath": ".default",
          "moduleName": "__fixtures__/ClassA",
          "state": "{\\"value\\":\\"value1\\"}",
        },
      ],
      "isPromise": false,
      "result": "Hello, world",
    },
  ],
}
`;

exports[`hello, world 3`] = `
Object {
  ".apply": Array [
    Object {
      "args": Array [
        "Hello, world",
        Object {
          "__$__": "function",
          "exportPath": ".addExclamation",
          "moduleName": "__fixtures__/module2",
        },
      ],
      "isPromise": false,
      "result": "Hello, world!",
    },
  ],
}
`;

exports[`hello, world 4`] = `
Object {
  ".default.getSeparator": Array [
    Object {
      "args": Array [
        Object {
          "__$__": "class",
          "exportPath": ".default",
          "moduleName": "__fixtures__/ClassA",
          "state": "{\\"value\\":\\"value1\\"}",
        },
        "comma",
      ],
      "isPromise": false,
      "result": ",",
    },
  ],
}
`;
